### ISSUE TITLE
RTSP failed with NOK in the UI

### ISSUE THREAD

**Issue by dobrishinov** [ThatUsernameAlreadyExist/TECKIN-TC100-Anyka-AK3918-camera-hacks#5]
Hey, thanks for the mod!
I tried the mod but the RTSP stream is failed.
Enabled log and here is the log for the v4l2rtspserver

```
Content of v4l2rtspserver.log

[ak_cmd_unregister_module:770] no server was register, can not unregister module
[m[1;31m
[do_match:225] file: /etc/jffs2/isp_f23_mipi2lane.conf not exist
[m
[ERROR]
	ak_vi_match_sensor failed
[ERROR]
	Terminate program on error. Config param 'abortonerror' set to 1
================== START ===============
/mnt/bin/v4l2rtspserver  video0,audio0 video1,audio1 -U root:pass -P 554
log level:500

[NOTICE]
	Version:  live555 version:2023.11.30
[NOTICE]
	Create V4L2 Source...video0
[NOTICE]
	default config: ) prefersharedconfig=0default log level=3
[1;31m
[ak_cmd_register_module:741] no server was register, can not register module
[m[1;33m[pcm_dac_set_volume:777] set DEV volume 5
[m## ChipIdval = 0x20160100
[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_module:770] no server was register, can not unregister module
[m[1;31m
[do_match:225] file: /etc/jffs2/isp_f23_mipi2lane.conf not exist
[m
[ERROR]
	ak_vi_match_sensor failed
[ERROR]
	Terminate program on error. Config param 'abortonerror' set to 1
================== START ===============
/mnt/bin/v4l2rtspserver  video0,audio0 video1,audio1 -U root:pass -P 554
log level:500

[NOTICE]
	Version:  live555 version:2023.11.30
[NOTICE]
	Create V4L2 Source...video0
[NOTICE]
	default config: ) prefersharedconfig=0default log level=3
[1;31m
[ak_cmd_register_module:741] no server was register, can not register module
[m[1;33m[pcm_dac_set_volume:777] set DEV volume 5
[m## ChipIdval = 0x20160100
[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_module:770] no server was register, can not unregister module
[m[1;31m
[do_match:225] file: /etc/jffs2/isp_f23_mipi2lane.conf not exist
[m
[ERROR]
	ak_vi_match_sensor failed
[ERROR]
	Terminate program on error. Config param 'abortonerror' set to 1
================== START ===============
/mnt/bin/v4l2rtspserver  video0,audio0 video1,audio1 -U root:pass -P 554
log level:500

[NOTICE]
	Version:  live555 version:2023.11.30
[NOTICE]
	Create V4L2 Source...video0
[NOTICE]
	default config: ) prefersharedconfig=0default log level=3
[1;31m
[ak_cmd_register_module:741] no server was register, can not register module
[m[1;33m[pcm_dac_set_volume:777] set DEV volume 5
[m## ChipIdval = 0x20160100
[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_module:770] no server was register, can not unregister module
[m[1;31m
[do_match:225] file: /etc/jffs2/isp_f23_mipi2lane.conf not exist
[m
[ERROR]
	ak_vi_match_sensor failed
[ERROR]
	Terminate program on error. Config param 'abortonerror' set to 1
================== START ===============
/mnt/bin/v4l2rtspserver  video0,audio0 video1,audio1 -U root:pass -P 554
log level:500

[NOTICE]
	Version:  live555 version:2023.11.30
[NOTICE]
	Create V4L2 Source...video0
[NOTICE]
	default config: ) prefersharedconfig=0default log level=3
[1;31m
[ak_cmd_register_module:741] no server was register, can not register module
[m[1;33m[pcm_dac_set_volume:777] set DEV volume 5
[m## ChipIdval = 0x20160100
[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_module:770] no server was register, can not unregister module
[m[1;31m
[do_match:225] file: /etc/jffs2/isp_f23_mipi2lane.conf not exist
[m
[ERROR]
	ak_vi_match_sensor failed
[ERROR]
	Terminate program on error. Config param 'abortonerror' set to 1
================== START ===============
/mnt/bin/v4l2rtspserver  video0,audio0 video1,audio1 -U root:pass -P 554
log level:500

[NOTICE]
	Version:  live555 version:2023.11.30
[NOTICE]
	Create V4L2 Source...video0
[NOTICE]
	default config: ) prefersharedconfig=0default log level=3
[1;31m
[ak_cmd_register_module:741] no server was register, can not register module
[m[1;33m[pcm_dac_set_volume:777] set DEV volume 5
[m## ChipIdval = 0x20160100
[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_module:770] no server was register, can not unregister module
[m[1;31m
[do_match:225] file: /etc/jffs2/isp_f23_mipi2lane.conf not exist
[m
[ERROR]
	ak_vi_match_sensor failed
[ERROR]
	Terminate program on error. Config param 'abortonerror' set to 1
================== START ===============
/mnt/bin/v4l2rtspserver  video0,audio0 video1,audio1 -U root:pass -P 554
log level:500

[NOTICE]
	Version:  live555 version:2023.11.30
[NOTICE]
	Create V4L2 Source...video0
[NOTICE]
	default config: ) prefersharedconfig=0default log level=3
[1;31m
[ak_cmd_register_module:741] no server was register, can not register module
[m[1;33m[pcm_dac_set_volume:777] set DEV volume 5
[m## ChipIdval = 0x20160100
[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_module:770] no server was register, can not unregister module
[m[1;31m
[do_match:225] file: /etc/jffs2/isp_f23_mipi2lane.conf not exist
[m
[ERROR]
	ak_vi_match_sensor failed
[ERROR]
	Terminate program on error. Config param 'abortonerror' set to 1
================== START ===============
/mnt/bin/v4l2rtspserver  video0,audio0 video1,audio1 -U root:pass -P 554
log level:500

[NOTICE]
	Version:  live555 version:2023.11.30
[NOTICE]
	Create V4L2 Source...video0
[NOTICE]
	default config: ) prefersharedconfig=0default log level=3
[1;31m
[ak_cmd_register_module:741] no server was register, can not register module
[m[1;33m[pcm_dac_set_volume:777] set DEV volume 5
[m## ChipIdval = 0x20160100
[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_module:770] no server was register, can not unregister module
[m[1;31m
[do_match:225] file: /etc/jffs2/isp_f23_mipi2lane.conf not exist
[m
[ERROR]
	ak_vi_match_sensor failed
[ERROR]
	Terminate program on error. Config param 'abortonerror' set to 1
================== START ===============
/mnt/bin/v4l2rtspserver  video0,audio0 video1,audio1 -U root:pass -P 554
log level:500

[NOTICE]
	Version:  live555 version:2023.11.30
[NOTICE]
	Create V4L2 Source...video0
[NOTICE]
	default config: ) prefersharedconfig=0default log level=3
[1;31m
[ak_cmd_register_module:741] no server was register, can not register module
[m[1;33m[pcm_dac_set_volume:777] set DEV volume 5
[m## ChipIdval = 0x20160100
[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_module:770] no server was register, can not unregister module
[m[1;31m
[do_match:225] file: /etc/jffs2/isp_f23_mipi2lane.conf not exist
[m
[ERROR]
	ak_vi_match_sensor failed
[ERROR]
	Terminate program on error. Config param 'abortonerror' set to 1
```

Here is the summary:
```
Summary of all log files:

==> /var/log/ak_print_level <==
3
==> /var/log/lighttpd-cgi-stderr.log <==

==> /var/log/lighttpd-error.log <==
2024-03-18 10:27:20: (server.c.1704) server started (lighttpd/1.4.69)
2024-03-18 10:27:51: (mod_openssl.c.3001) SSL: 1 -1 error:0A000416:SSL routines::sslv3 alert certificate unknown
2024-03-18 10:27:51: (connections.c.439) connection closed: write failed on fd 10

==> /var/log/messages <==
Mar 18 10:27:59 anyka kern.debug kernel: @@@ [playback_ioctl:1274]: IOC_RSTBUF @@@
Mar 18 10:27:59 anyka kern.info kernel: @@@ [playback_close]: enter @@@
Mar 18 10:27:59 anyka kern.debug kernel: @@@ [playback_pause:998]: playback is not working now @@@
Mar 18 10:27:59 anyka kern.debug kernel: @@@ [set_channel_source:105]: addr=0, src=0 @@@
Mar 18 10:27:59 anyka kern.warn kernel: ak39_set_hp_power 0,soft_de_pipa=1
Mar 18 10:27:59 anyka kern.debug kernel: @@@ [set_channel_source:153]: src=0, s_dac=0, s_linein=0, s_mic=1 @@@
Mar 18 10:27:59 anyka kern.debug kernel: @@@ [playback_close:1631]: OK @@@
Mar 18 10:27:59 anyka user.err syslog:  [ak_cmd_unregister_msg_handle:687] no server was register, can not register message
Mar 18 10:27:59 anyka user.err syslog:  [ak_cmd_unregister_module:770] no server was register, can not unregister module
Mar 18 10:27:59 anyka user.err syslog:  [do_match:225] file: /etc/jffs2/isp_f23_mipi2lane.conf not exist

==> /var/log/service-watchdog.log <==
Mon Mar 18 10:27:40 UTC 2024 Service /mnt/controlscripts/rtsp-h26x not running, error count: 1
/mnt/controlscripts/rtsp-h26x - perform restart
Mon Mar 18 10:27:57 UTC 2024 Service /mnt/controlscripts/rtsp-h26x not running, error count: 2
/mnt/controlscripts/rtsp-h26x - perform restart

===SD card logs===

==> /mnt/log/startup.log <==
Autostart...
Run /mnt/config/autostart/00_system-config
Run /mnt/config/autostart/01_system-emergency-telnet
Run /mnt/config/autostart/02_system-webserver
Run /mnt/config/autostart/auto-night-detection
Run /mnt/config/autostart/blue-led
Run /mnt/config/autostart/onvif
Run /mnt/config/autostart/rtsp-h26x
Mon Mar 18 10:27:24 UTC 2024
--------Starting Hacks Finished!--------

==> /mnt/log/v4l2rtspserver.log <==
[ak_cmd_unregister_msg_handle:687] no server was register, can not register message
[m[1;31m
[ak_cmd_unregister_module:770] no server was register, can not unregister module
[m[1;31m
[do_match:225] file: /etc/jffs2/isp_f23_mipi2lane.conf not exist
[m
[ERROR]
	ak_vi_match_sensor failed
[ERROR]
	Terminate program on error. Config param 'abortonerror' set to 1
```

**Comment by ThatUsernameAlreadyExist** [comment]
According logs your camera have different sensor. See this: https://github.com/ThatUsernameAlreadyExist/TECKIN-TC100-Anyka-AK3918-camera-hacks/issues/1
You must set correct sensor file name in rtspserver.conf file.

**Comment by dobrishinov** [comment]
Thank you for the reference!
My camera is the same but with another brand - Xbro XC100.

The hardware it's seems to be the same as the https://github.com/ThatUsernameAlreadyExist/TECKIN-TC100-Anyka-AK3918-camera-hacks/issues/1 issue.

![image](https://github.com/ThatUsernameAlreadyExist/TECKIN-TC100-Anyka-AK3918-camera-hacks/assets/10928366/6a735af7-67b5-4475-8dc4-dbe1daa12f35)

The fix is easy. I run the ftp server on the camera, checked the file in /etc/jffs2/ and I found that I have /isp_f37_mipi1lane.conf
Then I changed in the rtspserver.conf in the SD card sensor=/etc/jffs2/isp_f23_mipi2lane.conf to be sensor=/etc/jffs2/isp_f37_mipi1lane.conf and everything is running perfectly!

Thanks a lot for the great mod!

PS. Two suggestions:
- Can we check automatically which sensor is available and to be set automatically or at least some option in the UI on which easily we can put the new sensor type instead to edit the rtspserver.config.
- I saw that when we click on the menu (Settings/Services, etc) there is some delay for the response, it will be nice to have some loading spinner or something that preventing user to click again before the request is finished.
